lib/msf/ui/console/command_dispatcher/core.rb

```diff
@@ -1241,6 +1265,52 @@ def cmd_save(*args)
     print_line("Saved configuration to: #{Msf::Config.config_file}")
   end
 
+  # Reload default options for the currently active module
+  def cmd_save_reload
+    active_module.import_defaults
+    print_line('Reloaded default options.')
+  end
+
+  # Load the currently active module options from the saved config file.
+  def cmd_save_load
+    active_module.load_config
+    print_line('Loaded previously saved options.')
+  end
+
+  #
+  # Deletes or saves the active datastore contents to disk for automatic use across
+  # restarts of the console.
+  #
+  def cmd_save(*args)
+    if args.include?('-h') || args.include?('--help')
+      cmd_save_help
+      return
+    end
+
+    if args.empty?
+      # Save config for current module
+      cmd_save_to_file
+    end
+
+    @@save_opts.parse(args) do |opt|
+      case opt
+      when '-D'
+        # Delete all saved module info
+        cmd_save_delete_all_modules_configs
+      when '-d'
+        # Delete saved info for current module only
+        cmd_save_delete
+      when '-r'
+        cmd_save_reload
```

Comment:
I wonder if it would be nicer to inline these functions, not sure if it's worth the overhead of having separate methods here 👀